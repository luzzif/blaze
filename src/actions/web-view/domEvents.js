export const DOM_READY = "DOM_READY";
export const domReady = () => ({
    type: DOM_READY
});

export const LOAD_COMMIT = "LOAD_COMMIT";
export const loadCommit = (url, isMainFrame) => ({
    type: LOAD_COMMIT,
    url,
    isMainFrame
});

export const DID_START_LOADING = "DID_START_LOADING";
export const DID_STOP_LOADING = "DID_STOP_LOADING";
export const DID_FINISH_LOAD = "DID_FINISH_LOAD";
export const DID_FRAME_FINISH_LOAD = "DID_FRAME_FINISH_LOAD";
export const DID_FAIL_LOAD = "DID_FAIL_LOAD";

export const didStartLoading = () => ({
    type: DID_START_LOADING
});
export const didStopLoading = () => ({
    type: DID_STOP_LOADING
});
export const didFinishLoad = () => ({
    type: DID_FINISH_LOAD
});
export const didFrameFinishLoad = isMainFrame => ({
    type: DID_FRAME_FINISH_LOAD,
    isMainFrame
});
export const didFailLoad = (
    errorCode,
    errorDescription,
    validatedURL,
    isMainFrame
) => ({
    type: DID_FAIL_LOAD,
    errorCode,
    errorDescription,
    validatedURL,
    isMainFrame
});

export const DID_GET_RESPONSE_DETAILS = "DID_GET_RESPONSE_DETAILS";
export const didGetResponseDetails = (
    status,
    newUrl,
    originalUrl,
    httpResponseCode,
    requestMethod,
    referrer,
    headers,
    resourceType
) => ({
    type: DID_GET_RESPONSE_DETAILS,
    status,
    newUrl,
    originalUrl,
    httpResponseCode,
    requestMethod,
    referrer,
    headers,
    resourceType
});

export const DID_GET_REDIRECT_REQUEST = "DID_GET_REDIRECT_REQUEST";
export const didGetRedirectRequest = (oldUrl, newUrl, isMainFrame) => ({
    type: DID_GET_REDIRECT_REQUEST,
    oldUrl,
    newUrl,
    isMainFrame
});

export const PAGE_TITLE_UPDATED = "PAGE_TITLE_UPDATED";
export const PAGE_FAVICON_UPDATED = "PAGE_FAVICON_UPDATED";

export const pageTitleUpdated = (title, explicitSet) => ({
    type: PAGE_TITLE_UPDATED,
    title,
    explicitSet
});
export const pageFaviconUpdated = favicons => ({
    type: PAGE_FAVICON_UPDATED,
    favicons
});

export const ENTER_HTML_FULL_SCREEN = "ENTER_HTML_FULL_SCREEN";
export const LEAVE_HTML_FULL_SCREEN = "LEAVE_HTML_FULL_SCREEN";

export const enterHtmlFullScreen = () => ({
    type: ENTER_HTML_FULL_SCREEN
});
export const leaveHtmlFullScreen = () => ({
    type: LEAVE_HTML_FULL_SCREEN
});

export const CONSOLE_MESSAGE = "CONSOLE_MESSAGE";
export const consoleMessage = (level, message, line, sourceId) => ({
    type: CONSOLE_MESSAGE,
    level,
    message,
    line,
    sourceId
});

export const FOUND_IN_PAGE = "FOUND_IN_PAGE";
export const foundInPage = ({
    requestId,
    activeMatchOrdinal,
    matches,
    selectionArea,
    finalUpdate
}) => ({
    type: FOUND_IN_PAGE,
    requestId,
    activeMatchOrdinal,
    matches,
    selectionArea,
    finalUpdate
});

export const NEW_WINDOW = "NEW_WINDOW";
export const newWindow = (url, frameName, disposition, options) => ({
    type: NEW_WINDOW,
    url,
    frameName,
    disposition,
    options
});

export const WILL_NAVIGATE = "WILL_NAVIGATE";
export const DID_NAVIGATE = "DID_NAVIGATE";
export const DID_NAVIGATE_IN_PAGE = "DID_NAVIGATE_IN_PAGE";

export const willNavigate = url => ({
    type: WILL_NAVIGATE,
    url
});
export const didNavigate = url => ({
    type: DID_NAVIGATE,
    url
});
export const didNavigateInPage = (isMainFrame, url) => ({
    type: DID_NAVIGATE_IN_PAGE,
    isMainFrame,
    url
});

export const CLOSE = "CLOSE";
export const close = () => ({
    type: CLOSE
});

export const IPC_MESSAGE = "IPC_MESSAGE";
export const ipcMessage = (channel, args) => ({
    type: IPC_MESSAGE,
    channel,
    args
});

export const CRASHED = "CRASHED";
export const crashed = () => ({
    type: CRASHED
});

export const GPU_CRASHED = "GPU_CRASHED";
export const gpuCrashed = () => ({
    type: GPU_CRASHED
});

export const PLUGIN_CRASHED = "PLUGIN_CRASHED";
export const pluginCrashed = (name, version) => ({
    type: PLUGIN_CRASHED,
    name,
    version
});

export const DESTROYED = "DESTROYED";
export const destroyed = () => ({
    type: DESTROYED
});

export const MEDIA_STARTED_PLAYING = "MEDIA_START_PLAYING";
export const mediaStartedPlaying = () => ({
    type: MEDIA_STARTED_PLAYING
});

export const MEDIA_PAUSED = "MEDIA_PAUSED";
export const mediaPaused = () => ({
    type: MEDIA_PAUSED
});

export const DID_CHANGE_THEME_COLOR = "DID_CHANGE_THEME_COLOR";
export const didChangeThemeColor = themeColor => ({
    type: DID_CHANGE_THEME_COLOR,
    themeColor
});

export const UPDATE_TARGET_URL = "UPDATE_TARGET_URL";
export const updateTargetUrl = url => ({
    type: UPDATE_TARGET_URL,
    url
});

export const DEV_TOOLS_OPENED = "DEV_TOOLS_OPENED";
export const DEV_TOOLS_FOCUSED = "DEV_TOOLS_FOCUSED";
export const DEV_TOOLS_CLOSED = "DEV_TOOLS_CLOSED";

export const devToolsOpened = () => ({
    type: DEV_TOOLS_OPENED
});

export const devToolsFocused = () => ({
    type: DEV_TOOLS_FOCUSED
});

export const devToolsClosed = () => ({
    type: DEV_TOOLS_CLOSED
});
