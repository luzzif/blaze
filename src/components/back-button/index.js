import React from "react";
import { IconButton, withStyles } from "@material-ui/core";
import { ArrowBack } from "@material-ui/icons";
import { compose } from "redux";
import { styles } from "./styles";
import PropTypes from "prop-types";

const BasicBackButton = ({ classes, ...rest }) => (
    <IconButton {...rest} className={classes.root}>
        <ArrowBack />
    </IconButton>
);

BasicBackButton.propTypes = {
    classes: PropTypes.object.isRequired
};

const enhanceComponent = compose(withStyles(styles));

export const BackButton = enhanceComponent(BasicBackButton);
