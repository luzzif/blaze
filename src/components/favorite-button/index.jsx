import React from "react";
import { IconButton, withStyles } from "@material-ui/core";
import { Favorite } from "@material-ui/icons";
import { compose } from "redux";
import { styles } from "./styles";
import PropTypes from "prop-types";

const BasicFavoriteButton = ({ classes, ...rest }) => (
    <IconButton {...rest} className={classes.root}>
        <Favorite />
    </IconButton>
);

BasicFavoriteButton.propTypes = {
    classes: PropTypes.object.isRequired
};

const enhanceComponent = compose(withStyles(styles));

export const FavoriteButton = enhanceComponent(BasicFavoriteButton);
