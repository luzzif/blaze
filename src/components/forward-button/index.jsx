import React from "react";
import { IconButton, withStyles } from "@material-ui/core";
import { ArrowForward } from "@material-ui/icons";
import { compose } from "redux";
import { styles } from "./styles";
import PropTypes from "prop-types";

const BasicForwardButton = ({ classes, ...rest }) => (
    <IconButton {...rest} className={classes.root}>
        <ArrowForward />
    </IconButton>
);

BasicForwardButton.propTypes = {
    classes: PropTypes.object.isRequired
};

const enhanceComponent = compose(withStyles(styles));

export const ForwardButton = enhanceComponent(BasicForwardButton);
