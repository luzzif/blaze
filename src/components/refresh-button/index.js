import React from "react";
import { IconButton, withStyles } from "@material-ui/core";
import { Refresh } from "@material-ui/icons";
import { compose } from "redux";
import { styles } from "./styles";
import PropTypes from "prop-types";

const BasicRefreshButton = ({ classes, ...rest }) => (
    <IconButton {...rest} className={classes.root}>
        <Refresh />
    </IconButton>
);

BasicRefreshButton.propTypes = {
    classes: PropTypes.object.isRequired
};

const enhanceComponent = compose(withStyles(styles));

export const RefreshButton = enhanceComponent(BasicRefreshButton);
