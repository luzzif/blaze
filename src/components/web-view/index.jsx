import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core";
import { styles } from "./styles";
import { compose } from "redux";

const BasicWebView = ({ classes, ...rest }) => (
    <webview className={classes.root} src="http://www.github.com" />
);

BasicWebView.propTypes = {
    classes: PropTypes.object.isRequired
};

const enhanceComponent = compose(withStyles(styles));

export const WebView = enhanceComponent(BasicWebView);
