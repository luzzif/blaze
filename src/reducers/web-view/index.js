const defaultState = {};

export const webViewsReducer = (state = defaultState, { type, payload }) => {
    switch (type) {
        default: {
            return { ...state };
        }
    }
};
