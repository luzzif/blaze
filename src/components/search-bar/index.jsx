import React from "react";
import PropTypes from "prop-types";
import { Card, CardContent, withStyles, Grid } from "@material-ui/core";
import { compose } from "redux";
import { styles } from "./styles";
import { FavoriteButton } from "../favorite-button";

const BasicSearchBar = ({ classes, ...rest }) => (
    <Card {...rest} classes={{ root: classes.root }}>
        <CardContent classes={{ root: classes.content }}>
            <div className={classes.container}>
                <input className={classes.input} type="text" />
                <FavoriteButton />
            </div>
        </CardContent>
    </Card>
);

BasicSearchBar.propTypes = {
    classes: PropTypes.object.isRequired
};

const enhanceComponent = compose(withStyles(styles));

export const SearchBar = enhanceComponent(BasicSearchBar);
