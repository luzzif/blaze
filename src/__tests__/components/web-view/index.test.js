import React from "react";
import { WebView } from "./../../../components/web-view";
import TestRenderer from "react-test-renderer";

describe("web view", () => {
    let testInstance = null;

    beforeEach(() => {
        testInstance = TestRenderer.create(<WebView />);
    });

    it("should be rendered", () => {
        expect(testInstance.toJSON().type).toBe("webview");
        expect(testInstance).toMatchSnapshot();
    });
});
