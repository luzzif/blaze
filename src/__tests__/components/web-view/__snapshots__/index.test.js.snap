exports[`back button should be clickable 1`] = `
<button
  className="MuiButtonBase-root-8 MuiIconButton-root-2 BasicSettingsButton-root-1"
  disabled={false}
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button">
  <span
    className="MuiIconButton-label-7">
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-11"
      color={undefined}
      focusable="false"
      viewBox="0 0 24 24">
      <path
        d="M0 0h24v24H0z"
        fill="none" />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z" />
    </svg>
  </span>
  <span
    className="MuiTouchRipple-root-18" />
</button>
`;

exports[`back button should contain a menu icon 1`] = `
<button
  className="MuiButtonBase-root-8 MuiIconButton-root-2 BasicSettingsButton-root-1"
  disabled={false}
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button">
  <span
    className="MuiIconButton-label-7">
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-11"
      color={undefined}
      focusable="false"
      viewBox="0 0 24 24">
      <path
        d="M0 0h24v24H0z"
        fill="none" />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z" />
    </svg>
  </span>
  <span
    className="MuiTouchRipple-root-18" />
</button>
`;

exports[`web view button should be rendered 1`] = `
<webview
  className="BasicWebView-root-1"
  src="http://www.github.com" />
`;

exports[`web view should be rendered 1`] = `
<webview
  className="BasicWebView-root-1"
  src="http://www.github.com" />
`;
